packages/portal/src/index.js

```diff
@@ -4,17 +4,25 @@ import Component from "@reach/component-component";
 
 let Portal = ({ children, type = "reach-portal" }) => (
   <Component
-    getRefs={() => ({ node: null })}
+    getRefs={() => ({ mountNode: null, portalNode: null })}
     didMount={({ refs, forceUpdate }) => {
-      refs.node = document.createElement(type);
-      document.body.appendChild(refs.node);
+      // It's possible that the content we are portal has, itself, been portaled.
+      // In that case, it's important to append to the correct document element.
+      const ownerDocument = refs.mountNode.ownerDocument;
```

Comment:
Maybe there's a way for a component to get the `ownerDocument` without going through the creation of a ref. I'll have to dig into ReactDOM's source tomorrow and see.

Reply:
from your experience do you think that it is better to append to the body or is this approach fine https://github.com/giuseppeg/react-layers-manager/blob/master/src/LayersManager.js#L13-L16 ?

Reply:
TBH, I'm not familiar with rhe isolation property. Looks like it's not supported by IE/Edge though so likely a deal breaker there.

Reply:
oh I didn't mean that exact piece of code, but yea isolation creates a stacking context without affecting layout and the provider is just wrapping the app to trap any stacking context in the app. Then it creates a separate one where it appends the <Layer> (portals) instances. I'll fix the public repo to support ie/edge

I was wondering if, because of other 3rd parties libs, it'd be better to instead append to the body all the time

Reply:
I don't know. I haven't done a lot of work with portals